### PR TITLE
Design WelcomePage

### PR DESCRIPTION
**Pull request by law-levi-san**

### CODE REVIEW

File: styles.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const style = StyleSheet.create({
+  container: {
+    backgroundColor: "#FFF7E6",
+    paddingVertical: 50,
+    paddingHorizontal: 20,
+    justifyContent: "center",
+  },
+  buttonStyle: {
+    // TODO: Add generic button style for touchable opacity component
+    backgroundColor: "#FFC1C1",
```
Comment (Kicchu02) [inline styles.tsx]: Remove comments.

File: components/WelcomePage.tsx [hunk 2]
```diff
@@ -1,10 +1,26 @@
 import React from "react";
-import { Text, View } from "react-native";
+import {
+  ImageBackground,
+  ScrollView,
+  Text,
```
Comment (Kicchu02) [inline components/WelcomePage.tsx]: Optimise the imports.

File: assets/bg.jpg
Comment (Kicchu02) [inline assets/bg.jpg:1]: Where is this image used?

File: styles.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    marginTop: 20,
+    margin: 40,
+    padding: 20,
+    borderRadius: 40,
+  },
+  butonTextStyle: {
+    //TODO
+    color: "#4A4A4A", // dark grey
+    textAlign: "center",
```
Comment (Kicchu02) [inline styles.tsx]: I would suggest you to creat a `Theme.tsx` file in the `types/` folder that holds a bunch of colour and other property values. This way you can reuse them throughout the project and maintain theme consistency. Example:

```js
Color {
    const DARK_GRAY = "#4a4a4a"
}
```
This way, you can directly use `Color.DARK_GRAY` wherever needed.

Decide which data structure is best for this approach. Either named enums or objects or something else.

File: styles.tsx
Comment (Kicchu02) [inline styles.tsx:1]: IMO we should rename this file as `Styles.tsx`.

**Review by Kicchu02 — CHANGES_REQUESTED** [review]
image

Please fix this bottom white part. The view is not truly centre-aligned or full-screened.

**Comment by law-levi-san** [comment]
> image
> 
> Please fix this bottom white part. The view is not truly centre-aligned or full-screened.

I have, check once pls

File: styles.tsx (hunk 3 above)
Comment (law-levi-san) [reply styles.tsx]: I have created a file `Theme.tsx'  to store colors as directed and I have used Objects for this. Check once pls

File: assets/bg.jpg
Comment (law-levi-san) [reply assets/bg.jpg:1]: I have removed the image from assests folder since it is not required. Pls check

File: components/WelcomePage.tsx (hunk 2 above)
Comment (law-levi-san) [reply components/WelcomePage.tsx]: I have, look into it once.

File: styles.tsx (hunk 1 above)
Comment (law-levi-san) [reply styles.tsx]: I have removed the comments in the recent commit.

File: styles.tsx
Comment (law-levi-san) [reply styles.tsx:1]: I have renamed the file but I faced issue while importing it in WelcomePage.tsx as it wasn't taking `Styles.tsx` as a valid import statement. Guide me in this pls. I have left the import statement as it was i.e `styles.tsx` .

Comment (Kicchu02) [reply styles.tsx:1]: The shortcut for refactoring is f2 in vscode. Try renaming that way and it should prompt you to refactor all usages in the project.

File: styles.tsx (hunk 3 above)
Comment (Kicchu02) [reply styles.tsx]: It's alright for now. Although I was thinking of a more versatile approach. Maybe when the need arises, you should be ready to refactor.

**Comment by Kicchu02** [comment]
> image
> 
> Please fix this bottom white part. The view is not truly centre-aligned or full-screened.

This part is still not fixed. Please explain the approach you have taken to fix this.

File: components/WelcomePage.tsx [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
+      <Text style={style.textHeaderStyle}>WELCOME TO STATES APP!</Text>
+      <TouchableOpacity style={style.buttonStyle}>
+        <Text style={style.butonTextStyle}>Click on the correct state</Text>
+      </TouchableOpacity>
+      <TouchableOpacity style={style.buttonStyle}>
+        <Text style={style.butonTextStyle}>Enter correct state name</Text>
+      </TouchableOpacity>
+      <TouchableOpacity style={style.buttonStyle}>
+        <Text style={style.butonTextStyle}>Quiz on states</Text>
+      </TouchableOpacity>
```
Comment (Kicchu02) [inline components/WelcomePage.tsx]: This feels repeated with only the text changing. Create a separate component. Make sure you place that component in a proper hierarchy and structure. Just like the `types` folder. Then use that component everywhere in the project.

**Review by Kicchu02 — CHANGES_REQUESTED** [review]

**Review by Kicchu02 — APPROVED** [review]

**Pull request merged**